File: ServiceRouterProcessor/src/main/java/com/haha/servicerouterprocess/apt/Logger.java
package com.haha.servicerouterprocess.apt;

/**
 * @auther: haha
 * @Date: 2026/1/3
 * @Detail:
 */
public class Logger {

    private static final String PREFIX = "[DOF_ROUTER_APT] ";

    public static void debug(String s, Object... args) {
        System.out.println(format(s, args));
    }

    public static void info(String s, Object... args) {
        System.out.println(format(s, args));
    }

    public static void warn(String s, Object... args) {
        System.err.println(format(s, args));
    }

    public static void error(String s, Object... args) {
        System.err.println(format(s, args));
    }

    public static void error(Throwable t) {
        t.printStackTrace();
    }

    public static void fatal(Throwable t) {
        t.printStackTrace();
    }

    public static void fatal(String s, Object... args) {
        fatal(new RuntimeException(format(s, args)));
    }

    private static String format(String s, Object... args) {
        return PREFIX + (args.length == 0 ? s : String.format(s, args));
    }
}

File: ServiceRouterUtils/src/main/java/com/haha/servicerouterutils/utils/Logger.kt
package com.haha.servicerouterutils.utils

import android.util.Log

/**
 * @auther: haha
 * @Date:   2026/1/3
 * @Detail:
 */

object Logger {

    private val PREFIX = "[DOFRouter]::"
    private var isDebug = false

    fun openDebug() {
        isDebug = true
    }

    fun d(msg: String, throwable: Throwable? = null) {
        if (isDebug) {
            Log.d(PREFIX, msg, throwable)
        }
    }

    fun i(msg: String, throwable: Throwable? = null) {
        if (isDebug) {
            Log.i(PREFIX, msg, throwable)
        }
    }

    fun w(msg: String, throwable: Throwable? = null) {
        Log.w(PREFIX, msg, throwable)
    }

    fun wtf(msg: String, throwable: Throwable? = null) {
        Log.wtf(PREFIX, msg, throwable)
    }

    fun e(msg: String, throwable: Throwable? = null) {
        Log.e(PREFIX, msg, throwable)
    }
}